server/views/pages/goal/update/index.njk

```diff
@@ -154,11 +161,26 @@
                 ],
                 errorMessage: errors | findError('steps[' + (loop.index-1) + '][targetDateRange]')
               }) }}
-
+              
+              {% if loop.length > 1 %}
+                {{ govukButton({
+                  id: "delete-step-" + (loop.index-1),
+                  attributes: {
+                    'data-qa': 'step-' + (loop.index-1) + '-remove-button'
+                  },
+                  name: "action",
+                  value: 'delete-step-[' + step.reference + ']',
```

Comment:
`step.reference` wont have a value in the case of a new step being added, but not saved to the database (eg: user adds a step then immediately wants to delete it)
The `step.stepNumber` would be better (which will have a value, even if not persisted to the database) - though I'm not convinced that including it dynamically in the button action is the way to go (though I've not thought of a better way yet)

Reply:
Like I said, I'm not 100% on referencing the step in some way in the button action; but I can't think of a better way of doing it (without javascript!)
So, lets go with your approach, but use the `stepNumber` because `stepReferrence` wont always be set/present
```
value: 'delete-step-[' + step.stepNumber + ']',
```